src/main/java/cz/muni/fi/pa165/bluebat/service/CustomerServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package cz.muni.fi.pa165.bluebat.service;/* created by rudolf */
+
+import cz.muni.fi.pa165.bluebat.dao.CustomerDao;
+import cz.muni.fi.pa165.bluebat.entity.Customer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDate;
+
+/**
+ * @author rudolf
+ */
+@Service
+@Transactional
+public class CustomerServiceImpl implements CustomerService{
+
+    @Autowired
+    private final CustomerDao customerDao;
+
+    public CustomerServiceImpl(CustomerDao customerDao) {
+        this.customerDao = customerDao;
+    }
```

Comment:
@Autowired by měl být označen ten konstruktor, ale líbí se mi, že jsi zvolil dependecy injection přes konstruktor

Reply:
Ja som pochopil ze Autowired automaticky najde implemtaciu, myslim si ze to tak mal aj v prednaske.

Reply:
Ano, to najde. A fungovat budou oba způsoby. To, jak to máš napsané, je field injection, pokud dás Autowired místo toho nad konstruktor, bude z toho constructor injection. Obecně se doporučuje právě spíš constructor injection (zmiňovali to i na přednášce), proto by ten Autowired měl být spíš nad ním.

Reply:
Ano, to najde. A fungovat budou oba způsoby. To, jak to máš napsané, je field injection, pokud dás Autowired místo toho nad konstruktor, bude z toho constructor injection. Obecně se doporučuje právě spíš constructor injection (zmiňovali to i na přednášce), proto by ten Autowired měl být spíš nad ním.